app/models/piece.rb

```diff
@@ -35,8 +35,6 @@ def check_mate?
   def valid_move?(new_row, new_column)
     return false if out_of_boundary?(new_row, new_column)
     return false if no_move?(new_row, new_column)
-    #return false unless proper_length?(new_row, new_column)
-    true
```

Comment:
I agree. The super valid_move? already has return true statement, and not having it in the valid_move? in piece can make the code a bit DRYer.